### PR TITLE
TRUNK-247: Implemented constructors with required fields for some cor…

### PR DESCRIPTION
**Pull request by alinmihaila99**
…e classes

## Issue I worked on

see https://issues.openmrs.org/browse/TRUNK-247

## Checklist: I completed these to help reviewers :)

- [x] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

  No? -> [read here](https://wiki.openmrs.org/display/docs/Pull+Request+Tips) on how to squash multiple commits into one

- [x] My IDE is configured to follow the [**code style**](https://wiki.openmrs.org/display/docs/Java+Conventions) of this project.

  No? Unsure? -> [configure your IDE](https://wiki.openmrs.org/display/docs/How-To+Setup+And+Use+Your+IDE), format the code and add the changes with `git add . && git commit --amend`

- [ ] I have **added tests** to cover my changes. (If you refactored
  existing code that was well tested you do not have to add tests)

  No? -> write tests and add them to this commit `git add . && git commit --amend`

- [x] I ran `mvn clean package` right before creating this pull request and
  added all formatting changes to my commit.

  No? -> execute above command

- [x] All new and existing **tests passed**.

  No? -> figure out why and add the fix to your commit. It is your responsibility to make sure your code works.

- [x] My pull request is **based on the latest changes** of the master branch.

  No? Unsure? -> execute command `git pull --rebase upstream master`

### CODE REVIEW

File: api/src/main/java/org/openmrs/User.java [hunk 1]
```diff
@@ -77,6 +78,13 @@ public User(Person person) {
 		this.person = person;
 	}
 	
+	/** constructor with all required fields for Ugiser */
```
Comment (dkayiwa) [inline api/src/main/java/org/openmrs/User.java]: What is Ugiser?

**Comment by dkayiwa** [comment]
Did you deal with all objects in org.openmrs package?

File: api/src/main/java/org/openmrs/User.java (hunk 1 above)
Comment (alinmihaila99) [reply api/src/main/java/org/openmrs/User.java]: it meant User. I will resolve this

**Comment by alinmihaila99** [comment]
No, am still working on this issue, but I wanted to be sure what i did there is ok.

File: api/src/main/java/org/openmrs/User.java (hunk 1 above)
Comment (alinmihaila99) [reply api/src/main/java/org/openmrs/User.java]: it meant User. I corrected this

**Comment by alinmihaila99** [comment]
@dkayiwa  I think I implemented all of them. can you have a look, please?

**Comment by alinmihaila99** [comment]
@ibacher  can you also have a look at this, please?

File: api/src/main/java/org/openmrs/Drug.java [hunk 2]
```diff
@@ -64,7 +64,15 @@ public Drug() {
 	public Drug(Integer drugId) {
 		this.drugId = drugId;
 	}
-	
+
+	/** constructor with all required fields for Drug */
+	public Drug(Boolean combination, Concept concept, String name, User creator) {
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/Drug.java]: This surely at least needs `drugId`

File: api/src/main/java/org/openmrs/Field.java [hunk 3]
```diff
@@ -51,7 +51,15 @@ public Field() {
 	public Field(Integer fieldId) {
 		this.fieldId = fieldId;
 	}
-	
+
+	/** constructor with all required fields for Field */
+	public Field(FieldType fieldType, Boolean selectMultiple, User creator, String name) {
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/Field.java]: This should also take `fieldId`

File: api/src/main/java/org/openmrs/Form.java [hunk 4]
```diff
@@ -51,7 +51,14 @@ public Form() {
 	public Form(Integer formId) {
 		this.formId = formId;
 	}
-	
+
+	/** constructor fith required field for Form */
+	public Form(String name, User creator, String version) {
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/Form.java]: This should take `formId`

File: api/src/main/java/org/openmrs/Order.java [hunk 5]
```diff
@@ -148,6 +148,23 @@ public Order(Integer orderId) {
 		this.orderId = orderId;
 	}
 	
+	/** constructor with all required fields for Order */
+	public Order(Patient patient, OrderType orderType, Concept concept, Date dateActivated, Encounter encounter,
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/Order.java]: This will also need `orderId`

File: api/src/main/java/org/openmrs/Person.java [hunk 6]
```diff
@@ -182,6 +182,12 @@
 	public Person() {
 	}
 	
+	/** constructor with all required fields for Person */
+	public Person(Date personDateCreated, Boolean personVoided) {
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/Person.java]: This should have `personId`

File: api/src/main/java/org/openmrs/Program.java [hunk 7]
```diff
@@ -60,7 +60,14 @@ public Program(Integer programId) {
 	public Program(String name) {
 		setName(name);
 	}
-	
+
+	/** Constructor with all required fields for Program */
+	public Program(Concept concept, Concept outcomesConcept, Set<ProgramWorkflow> allWorkflows) {
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/Program.java]: This should have `programId`

File: api/src/main/java/org/openmrs/User.java [hunk 8]
```diff
@@ -77,6 +78,14 @@ public User(Person person) {
 		this.person = person;
 	}
 	
+	/** constructor with all required fields for User */
+	public User(Person person, String systemId,Map<String, String> userProperties, User creator){
```
Comment (ibacher) [inline api/src/main/java/org/openmrs/User.java]: This should take `userId`

Comment (alinmihaila99) [reply api/src/main/java/org/openmrs/User.java]: added userId

File: api/src/main/java/org/openmrs/Program.java (hunk 7 above)
Comment (alinmihaila99) [reply api/src/main/java/org/openmrs/Program.java]: added programId

File: api/src/main/java/org/openmrs/Person.java (hunk 6 above)
Comment (alinmihaila99) [reply api/src/main/java/org/openmrs/Person.java]: added personId

File: api/src/main/java/org/openmrs/Order.java (hunk 5 above)
Comment (alinmihaila99) [reply api/src/main/java/org/openmrs/Order.java]: added orderId

File: api/src/main/java/org/openmrs/Form.java (hunk 4 above)
Comment (alinmihaila99) [reply api/src/main/java/org/openmrs/Form.java]: added formId

File: api/src/main/java/org/openmrs/Field.java (hunk 3 above)
Comment (alinmihaila99) [reply api/src/main/java/org/openmrs/Field.java]: added fieldId

File: api/src/main/java/org/openmrs/Drug.java (hunk 2 above)
Comment (alinmihaila99) [reply api/src/main/java/org/openmrs/Drug.java]: added drugId

File: api/src/main/java/org/openmrs/Concept.java [hunk 9]
```diff
@@ -152,6 +152,23 @@ public Concept(Integer conceptId) {
...
+				   Boolean set){
+		super();
+		this.conceptId = conceptId;
+		this.conceptClass = conceptClass;
+		this.creator = creator;
+		this.retired = retired;
+		this.datatype = datatype;
+		this.conceptSets = conceptSets;
+		this.answers = answers;
+		this.conceptMappings = conceptMappings;
```
[the remark this answers is not in the document]
Comment (alinmihaila99) [reply api/src/main/java/org/openmrs/Concept.java]: deleted this

File: api/src/main/java/org/openmrs/Concept.java [hunk 10]
```diff
@@ -152,6 +152,23 @@ public Concept(Integer conceptId) {
...
+	public Concept(int conceptId, ConceptClass conceptClass, User creator, Boolean retired, ConceptDatatype datatype, 
+				   List <ConceptSet> conceptSets, List<ConceptAnswer> answers, List<ConceptMap> conceptMappings, 
+				   Boolean set){
+		super();
+		this.conceptId = conceptId;
+		this.conceptClass = conceptClass;
+		this.creator = creator;
+		this.retired = retired;
+		this.datatype = datatype;
+		this.conceptSets = conceptSets;
```
[the remark this answers is not in the document]
Comment (alinmihaila99) [reply api/src/main/java/org/openmrs/Concept.java]: deleted this

File: api/src/main/java/org/openmrs/Concept.java [hunk 11]
```diff
@@ -152,6 +152,23 @@ public Concept(Integer conceptId) {
...
+		this.answers = answers;
```
[the remark this answers is not in the document]
Comment (alinmihaila99) [reply api/src/main/java/org/openmrs/Concept.java]: deleted this

File: api/src/main/java/org/openmrs/Concept.java [hunk 12]
```diff
@@ -152,6 +152,23 @@ public Concept(Integer conceptId) {
...
+		this.conceptMappings = conceptMappings;
+		dateCreated = new Date();
```
[the remark this answers is not in the document]
Comment (alinmihaila99) [reply api/src/main/java/org/openmrs/Concept.java]: yes, you are right. I deleted this

File: api/src/main/java/org/openmrs/Concept.java [hunk 13]
```diff
@@ -152,6 +152,23 @@ public Concept(Integer conceptId) {
 		this.conceptId = conceptId;
 	}
 	
+	/** constructor with all required fields for Concept */
+	public Concept(int conceptId, ConceptClass conceptClass, User creator, Boolean retired, ConceptDatatype datatype, 
+				   List <ConceptSet> conceptSets, List<ConceptAnswer> answers, List<ConceptMap> conceptMappings, 
+				   Boolean set){
+		super();
```
[the remark this answers is not in the document]
Comment (alinmihaila99) [reply api/src/main/java/org/openmrs/Concept.java]: I left there accidentaly. removed it

**Comment by alinmihaila99** [comment]
@ibacher I resolved what you suggested. Thank you very much for this review, I will add a commit after I add other constructors. And about the tests, where should I write them?

**Comment by alinmihaila99** [comment]
Sorry, I had some problems with this pull request. But now I think it's ok. I made the changes you suggested and added other constructors. Waiting for a revew @ibacher

File: api/src/main/java/org/openmrs/Allergies.java [hunk 14]
```diff
@@ -33,6 +33,12 @@
 	
 	private List<Allergy> allergies = new ArrayList<>();
 
+	/** default constructor */
+	public Allergies() {
+	}
+	
+
```
Comment (varung-31) [inline api/src/main/java/org/openmrs/Allergies.java]: Please remove extra spaces.

**Pull request closed without merging**